### DISCUSSION TITLE
Replayable GUi

### DISCUSSION THREAD

**Discussion by EdorianDark** [slint-ui/slint#2184] [General] [answered]
Here is an interesting blog in which someone describes his ideas for a Rust gui: https://poignardazur.github.io/2023/02/02/masonry-01-and-my-vision-for-rust-ui/

The idea about Replayable GUI sounds really interesting.

**Comment by ogoffart** [comment, marked as answer]
This is a great blog post.
We actually have thought about many of the points in the article.
We want to eventually provide a good testing framework for Slint, that includes replaying data. There is even an issue for that: https://github.com/slint-ui/slint/issues/464

We don't have yet a good public API for testing, but we already have an internal test driver. 
The test framework can send any input event as if it was the user. Also Slint can use a mocked time, so the test can test animation with a fake time. (So a test don't need to wait real time for animation to finish, but can just replay the whole thing almost instantly)
